charts/MapTooltip.tsx

```diff
@@ -71,6 +76,51 @@ export class MapTooltip extends React.Component<MapTooltipProps> {
         return lastVal ? this.sparkBarsDatumXAccessor(lastVal) : undefined
     }
 
+    @computed get renderSparkBars() {
+        const { chart } = this
+        return chart.hasChartTab && chart.isLineChart
+    }
+
+    @computed get darkestColorInColorScheme() {
+        const { colorScale } = this.chart.map.data
+        return colorScale.isColorSchemeInverted
+            ? first(colorScale.baseColors)
+            : last(colorScale.baseColors)
+    }
+
+    @computed get barColor() {
+        const { colorScale } = this.chart.map.data
+        return colorScale.singleColorScale && !colorScale.isCustomColors
+            ? this.darkestColorInColorScheme
+            : undefined
+    }
+
+    @computed get tooltipHeightByType() {
+        return {
+            sparkBar: -134,
+            dataValue: -102,
+            noData: -65
```

Comment:
This is ok, but it feels _hacky_ – the tooltip already knows exactly how big it is: https://github.com/owid/owid-grapher/blob/b3b70fbb8eb7fad2d02935d416effe562a88e133/charts/Tooltip.tsx#L76-L81

So ideally we shouldn't hardcode it. Different browsers can also render it differently, and result in a different height.

As I understand, the problem is that we want the tooltip to have a fixed bottom position, and extend as far as it needs to the top?

We already solve this horizontally for LineChart and AreaChart tooltips – they are "anchored" to an X position, but flip left or right depending on where space is available: https://github.com/owid/owid-grapher/blob/b3b70fbb8eb7fad2d02935d416effe562a88e133/charts/Tooltip.tsx#L36-L44

Could we achieve the same for these map tooltips? Maybe we can add a parameter to `Tooltip` to prefer to show up on top by default, and only swap to bottom if there isn't enough space? (e.g. like tippy's [`placement` parameter](https://atomiks.github.io/tippyjs/#placement))

Let me know if I'm misunderstanding the problem!